src/main/java/ma/hibernate/dao/impl/PhoneDaoImpl.java

```diff
@@ -0,0 +1,63 @@
+package ma.hibernate.dao.impl;
+
+import java.util.List;
+import java.util.Map;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
+import ma.hibernate.dao.AbstractDao;
+import ma.hibernate.dao.PhoneDao;
+import ma.hibernate.model.Phone;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+
+public class PhoneDaoImpl extends AbstractDao implements PhoneDao {
+    public PhoneDaoImpl(SessionFactory sessionFactory) {
+        super(sessionFactory);
+    }
+
+    @Override
+    public Phone create(Phone phone) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(phone);
+            transaction.commit();
+            return phone;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Cant add phone to DB " + phone, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public List<Phone> findAll(Map<String, String[]> params) {
+        try (Session session = factory.openSession()) {
+            CriteriaBuilder criteriaBuilder = session.getCriteriaBuilder();
+            CriteriaQuery<Phone> query = criteriaBuilder.createQuery(Phone.class);
+            Root<Phone> phoneRoot = query.from(Phone.class);
+            Predicate finalPredicate = criteriaBuilder.and();
+            for (Map.Entry<String, String[]> entry : params.entrySet()) {
+                CriteriaBuilder.In<String> stringPredicate =
+                        criteriaBuilder.in(phoneRoot.get(entry.getKey()));
+                for (String value : entry.getValue()) {
+                    stringPredicate.value(value);
+                }
+                finalPredicate = criteriaBuilder.and(finalPredicate, stringPredicate);
+            }
+            return session.createQuery(query.where(finalPredicate)).getResultList();
+        } catch (Exception e) {
+            throw new RuntimeException("Cant find all phones with params", e);
```

Comment:
```suggestion
            throw new RuntimeException("Cant find all phones with params " + params, e);
```

Reply:
![image](https://user-images.githubusercontent.com/73486745/128328529-3417344f-8199-4f42-83bf-8580321f9ede.png)
![image](https://user-images.githubusercontent.com/73486745/128328583-a9a7bfcc-83a5-493f-abe4-c60eab393d83.png)

Reply:
the ideal solution would be adding `params` content in readable view to exception message